reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Throttle.kt

```diff
@@ -11,16 +12,16 @@ fun <T> Observable<T>.throttle(windowMillis: Long): Observable<T> =
 
         subscribeSafe(
             object : ObservableObserver<T>, ObservableCallbacks<T> by observer {
-                private var lastTime = 0L
+                private val lastTime = AtomicReference(0L)
 
                 override fun onSubscribe(disposable: Disposable) {
                     disposableWrapper.set(disposable)
                 }
 
                 override fun onNext(value: T) {
                     val time = uptimeMillis
-                    if (time - lastTime >= windowMillis) {
-                        lastTime = time
+                    if (time - lastTime.value >= windowMillis) {
```

Comment:
Race condition. Probably can be fixed by:
```kotlin
...
val lastTimeValue = lastTime.value
if (time - lastTimeValue >= windowMillis) {
  lastTime.compareAndSet(lastTimeValue, time) {
    observer.onNext(value)
  }
}

```

Reply:
See comment above, there is only one upstream